components/datetime/src/skeleton/helpers.rs

```diff
@@ -317,6 +319,34 @@ fn adjust_pattern_field_lengths(fields: &[Field], pattern: &mut Pattern) {
     })
 }
 
+/// Alters given Pattern so that it will have a fractional second field if it was requested.
+///
+/// If the requested skeleton included both seconds and fractional seconds and the dateFormatItem
+/// skeleton included seconds but not fractional seconds, then the seconds field of the corresponding
+/// pattern should be adjusted by appending the locale’s decimal separator, followed by the sequence
+/// of ‘S’ characters from the requested skeleton.
+/// (see https://unicode.org/reports/tr35/tr35-dates.html#Matching_Skeletons)
+fn append_fractional_seconds(pattern: &mut Pattern, fields: &[Field]) {
+    if let Some(requested_field) = fields
+        .iter()
+        .find(|field| field.symbol == FieldSymbol::Second(fields::Second::FractionalSecond))
+    {
+        let mut items = pattern.items.to_vec();
```

Comment:
Thought: This is incurring an additional allocation. Nothing comes to mind to optimize it out. It does seem odd to have to go through the `vec` and `ZeroVec` allocations to do any mutation.

First the ZeroVec is turned into a vec: `self.iter().collect()`
Next it's converted back to a ZeroVec: `ZeroVec::alloc_from_slice`

Reply:
Yes, I think we're stuck with this... `ZeroVec` works best when backed by static data.